### ISSUE TITLE
Improve posting to iNat experience (home screen uploads) with spotty cell signal

### ISSUE THREAD

**Issue by albullington** [inaturalist/SeekReactNative#352]
**Is your feature request related to a problem? Please describe.**
It's quick and easy to upload multiple observations over wifi, and it's easy to get a "no internet" error message when there's no cell/wifi data at all. However, when uploading multiple observations with 1-2 bars of cell signal, the uploads appear to stall with no progress on the process bar. Usually the observations are successfully uploaded to iNat, but the photos will not upload until there is better service.

**Describe the solution you'd like**
There are a few things that could happen here:
- [ ] Give users an option to only upload via wifi instead of cell service. This would be especially helpful for users who don't pay for unlimited data and might end up paying money for our automatic uploads.
- [ ] Give users an option to cancel the upload temporarily
- [ ] Add a `setTimeout` race condition which will stop trying to upload observations after a certain number of milliseconds, so we can show an error screen instead of a non-advancing progress bar (similar to how we implemented the [species photo fetch](https://github.com/inaturalist/SeekReactNative/commit/dac3788f35a260db622585fcd23c3472d55bf08f) in the AR Camera/Match Screen)
- [ ] Move the upload code to an actual background process, using a library like [react-native-background-upload](https://github.com/Vydia/react-native-background-upload). This would allow users to keep uploading photos while Seek is not actively being used or when a user leaves the Seek home screen. This needs some exploration to make sure we would still be able to show the progress bar and store details about whether uploads succeeded or not in Realm while Seek is backgrounded.

**Describe alternatives you've considered**
Open to suggestions from @abhasinat or others

**Additional context**
Stuck progress bar, two bars of cell service. I was able to finish uploading by 7:30, which means it took a full 5 minutes to upload 18 observations (really, just the observation photos).

![IMG_9633](https://user-images.githubusercontent.com/35536439/122469701-1b873800-cf72-11eb-9a22-3c310a33eb10.PNG)

**Comment by abhasinat** [comment]
I like option 3. This way, a user will realize that the observations are trying to upload, but the lack of internet is the issue. 

>  Add a setTimeout race condition which will stop trying to upload observations after a certain number of milliseconds, so we can show an error screen instead of a non-advancing progress bar (similar to how we implemented the species photo fetch in the AR Camera/Match Screen)

We can use the same error state styling we have right now- with either the existing "Upload failed" or a new error "Internet connection unstable. Please try again later."

**Comment by albullington** [comment]
Okay, I added option 3. I'm not sure how long we can expect these API calls to take for users with slower wifi at home, so I set these timeouts at:

- 5 seconds for fetching the login token
- 5 seconds for uploading the observation
- 15 seconds for uploading the photo

1 second for the first two and 8+ seconds for the photo upload works consistently on my wifi, but I'm assuming I have faster-than-average speeds. I'm also assuming that most uploads will stall on the photo upload part of this flow when the connection is spotty.
